### PR TITLE
add support for blocking on downstream build completion

### PR DESCRIPTION
**Pull request by twpedersen**
It may be useful to gate current build success on
downstream build status for eg. downstream integration
tests. Add an option "block" to make the trigger step
block on downstream build completion. The "block-timeout"
value specifies how long we should wait for.

First time writing Go so be gentle :)

One bug remains: when setting `block-timeout` to eg. `10s`, the `p.BlockTimeout` is still printed as `1m0s`. This looks similar to
https://discourse.drone.io/t/plugins-downstream-timeout-not-honored/2736

### CODE REVIEW

**Comment by tboerger** [comment]
I think `wait` and `wait_timeout` are already doing that?

**Comment by twpedersen** [comment]
No, see https://github.com/drone-plugins/drone-downstream/issues/60

`wait` just waits until the current build finishes. `block` would wait until the triggered build finishes.

File: main.go [hunk 1]
```diff
@@ -92,6 +103,8 @@ func run(c *cli.Context) error {
 		Wait:           c.Bool("wait"),
 		Timeout:        c.Duration("timeout"),
 		LastSuccessful: c.Bool("last-successful"),
+		Block:		c.Bool("block"),
```
Comment (appleboy) [inline main.go]: gofmt

Comment (twpedersen) [reply main.go]: done, thanks.

**Comment by twpedersen** [comment]
Closing. The workflow this was supposed to support is not really possible (or probably even desirable) within drone.

**Pull request closed without merging**